pkg/gatewayserver/io/ws/sessions.go

```diff
@@ -0,0 +1,79 @@
+// Copyright © 2020 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ws
+
+import (
+	"context"
+	"sync"
+
+	"go.thethings.network/lorawan-stack/v3/pkg/errors"
+)
+
+// Session contains the session state for a single gateway.
+type Session struct {
+	ID int32
+}
+
+var (
+	errSessionNotFound      = errors.DefineNotFound("session_not_found", "session for uid `{uid}` not found")
```

Comment:
`uid` is not a user-facing term. You can probably just omit `uid` here because the context is clear from the error namespace, otherwise replace by `gateway`